compiler/AST/primitive.cpp

```diff
@@ -715,3 +716,12 @@ VarSymbol* newMemDesc(const char* str) {
   memDescsVec.add(s);
   return memDescVar;
 }
+
+VarSymbol* newMemDesc(Symbol* sym) {
```

Comment:
But it doesn't make sense to have memDescs for different variables. MemDescs are for types, right? It would be trivial to pass s->typeInfo()->symbol in callChplHereAlloc.